File: cpp/Laptop.cpp
#include <iostream>
#include <string>
using namespace std;

class Laptop{
private:
  string id;
  string nama;
  int harga;
  int stok;

public:
  // Constructor untuk instansiasi
  Laptop(string id, string nama, int harga, int stok) {
    this->id = id;
    this->nama = nama;
    this->harga = harga;
    this->stok = stok;
  }

  // Getter id & Setter nama
  string getId() {
    return id;
  }

  void setId(string id) {
    this->id = id;
  }

  // Getter nama & Setter nama
  string getNama() {
    return nama;
  }

  void setNama(string nama) {
    this->nama = nama;
  }

  // Getter harga & Setter harga
  int getHarga() {
    return harga;
  } 

  void setHarga(int harga) {
    if (harga >= 0) {
      this->harga = harga;
    }
  }

  // Getter stok
  int getStok() {
    return stok;
  }

  void setStok(int stok) {
    this->stok = stok;
  }

  void showData() {
    cout << "ID   : " << id   << "\n"
         << "|    Nama : " << nama << "\n"
         << "|    Harga: " << harga << "\n"
         << "|    Stok : " << stok << "\n";
}

  // Destruktor
  ~Laptop() {}
};

File: java/Laptop.java
public class Laptop {
    private String id;
    private String nama;
    private int harga;
    private int stok;

    // Constructor
    public Laptop(String id, String nama, int harga, int stok) {
        this.id = id;
        this.nama = nama;
        this.harga = harga;
        this.stok = stok;
    }

    // Getter & Setter
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public int getHarga() {
        return harga;
    }

    public void setHarga(int harga) {
        if (harga >= 0) {
            this.harga = harga;
        }
    }

    public int getStok() {
        return stok;
    }

    public void setStok(int stok) {
        this.stok = stok;
    }

    public void showData() {
        System.out.println("ID   : " + id);
        System.out.println("|    Nama : " + nama);
        System.out.println("|    Harga: " + harga);
        System.out.println("|    Stok : " + stok);
    }
}
